sklearn/utils/estimator_checks.py

```diff
@@ -415,9 +448,8 @@ def check_estimator_sparse_data(name, estimator_orig):
     with ignore_warnings(category=DeprecationWarning):
         estimator = clone(estimator_orig)
     y = multioutput_estimator_convert_y_2d(estimator, y)
-    for sparse_format in ['csr', 'csc', 'dok', 'lil', 'coo', 'dia', 'bsr']:
-        X = X_csr.asformat(sparse_format)
-        # catch deprecation warnings
+    for matrix in _generate_sparse_matrix(X_csr):
```

Comment:
if instead of a dict, `_generate_sparse_matrix` returned a tuple (e.g. `return x, y`) you could just use `for format, matrix in _generate_sparse_matrix(X_csr):` here which would be clearer and more succinct